[KB] Update diff scans article for consistency, add Jenkins with GH/multibranch (#1878)
semgrep/semgrep-docs: docs/kb/semgrep-ci/trigger-diff-scans-env-var.md

<TabItem value='azure'>

To add this configuration in Azure Pipelines, follow the general instructions provided in [Sample CI configurations: Azure Pipelines](/docs/semgrep-ci/sample-ci-configs#azure-pipelines). If your repository's default branch is not `main`, change the references to `main` to the name of your default branch.

```yaml
steps:
- checkout: self
  clean: true

>   fetchDepth: 20

Reviewer: lol 10000 😳

Author: hey, you put that there :)

persistCredentials: true
- script: |
    python -m pip install --upgrade pip
    pip install semgrep
    if [ $(System.PullRequest.PullRequestId) -ge 0 ]; then
      echo "Pull Request Scan from branch: $(Build.SourceBranchName)"
      git fetch origin main:origin/main
      export SEMGREP_PR_ID=$(System.PullRequest.PullRequestId)
      export SEMGREP_BASELINE_REF='origin/main'
      semgrep ci
```

> If you are also running full scans for the repository (recommended) you can either use if clauses or define separate templates for full scans and [diff-aware scans](/deployment/customize-ci-jobs#set-up-diff-aware-scans) in Azure Pipelines. Diff-aware scans require the use of the  `SEMGREP_PR_ID` and `SEMGREP_BASELINE_REF` variables, while full scans do not. Full scans would typically be run on the condition `if [ $(Build.SourceBranchName) = "main" ]`.

Reviewer: ```suggestion
If you are running both full and diff-aware scans for the repository, you can use if clauses or define separate templates for full scans and [diff-aware scans](/deployment/customize-ci-jobs#set-up-diff-aware-scans) in Azure Pipelines. Diff-aware scans require the use of the  `SEMGREP_PR_ID` and `SEMGREP_BASELINE_REF` variables, while full scans do not. Full scans are typically run on the condition `if [ $(Build.SourceBranchName) = "main" ]`.
```

</TabItem>
<TabItem value='gitlab'>

> Set up your `.gitlab-ci.yml` conditions (usually `rules`) to run a scan if `$CI_MERGE_REQUEST_IID` is defined. Semgrep automatically runs a diff-aware scan if the variable is present, as it is in merge request pipelines:

Reviewer: ```suggestion
Set up your `.gitlab-ci.yml` conditions (usually `rules`) to run a scan if `$CI_MERGE_REQUEST_IID` is defined. Semgrep automatically runs a diff-aware scan if the variable is present, as it does in merge request pipelines:
```

Not sure if I'm understanding this correctly?

Author: Yeah, I think there's a semantic nuance here. The variable **is** present in merge request pipelines, and therefore Semgrep does run a diff-aware scan there.
